File: Atividade5/BANCO-OO/src/classes/Conta.java
package classes;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

public class Conta {
    private int idConta;
    private String numero;
    private Cliente titular;
    private LocalDate dataAbertura;
    private double saldo;

    static DateTimeFormatter formatoBR = DateTimeFormatter.ofPattern("dd/MM/yyyy");

    public Conta(int idConta, String numero, Cliente titular, LocalDate dataAbertura) {
        this.idConta = idConta;
        this.numero = numero;
        this.titular = titular;
        this.dataAbertura = dataAbertura;
        this.saldo = 0.0;
    }
    
    public void sacar(double valor){
        if (this.saldo >= valor && this.saldo > 0){
            this.saldo -= valor;

            System.out.println("Saque realizado com sucesso!");
            System.out.printf("Saldo atualizado: R$ %.2f\n", this.saldo);
        }
        else{
            System.out.println("Saldo insuficiente para fazer o saque!");
        }
    }

    public void depositar(double valor){
        if(valor > 0){
            this.saldo += valor;

            System.out.println("Depósito realizado com sucesso!");
            System.out.printf("Saldo atualizado: R$ %.2f\n", this.saldo);
        }
    }

    public void transferir(Conta conta, double valor){
        if(this.saldo > 0 && valor <= this.saldo){
            this.sacar(valor);
            conta.depositar(valor);
        }
        else{
            System.out.println("Saldo insuficiente para fazer a transferência!");
        }
    }

    public void mostrarInfo(){
    System.out.printf(
    """
    | ID: %d
    | NÚMERO: %s
    | DATA ABERTURA: %s
    | TITULAR:
    %s
    | SALDO: %.2f
    ------------------------------------
    """, this.idConta, this.numero, this.dataAbertura.format(formatoBR), this.titular.dados(), this.saldo);
    }

    public int getIdConta() {
        return this.idConta;
    }

    public void setIdConta(int idConta) {
        this.idConta = idConta;
    }

    public String getNumero() {
        return this.numero;
    }

    public void setNumero(String numero) {
        this.numero = numero;
    }

    public Cliente getTitular() {
        return this.titular;
    }

    public void setTitular(Cliente titular) {
        this.titular = titular;
    }

    public double getSaldo() {
        return this.saldo;
    }

    public LocalDate getDataAbertura() {
        return this.dataAbertura;
    }

    public void setDataAbertura(LocalDate dataAbertura) {
        this.dataAbertura = dataAbertura;
    }
}

File: Atividade5-TS/sistema-bancario/src/classes/Conta.ts
import Cliente from "./Cliente";

export default class Conta {
  private _id: number;
  private _numero: string;
  private _saldo: number;
  private _titular: Cliente;
  private _dataAbertura: string;

  constructor(id: number, numero: string, cliente: Cliente, saldo: number) {
    this._id = id;
    this._numero = numero;
    this._titular = cliente;
    this._saldo = saldo;
    this._dataAbertura = new Date().toLocaleDateString("pt-BR");
  }

  public sacar(valor: number): void {
    if (valor > 0) {
      this._saldo -= valor;
      console.log(`Saldo atualizado: R$ ${this._saldo.toFixed()}`);
    }
  }

  public depositar(valor: number): void {
    if (valor > 0) {
      this._saldo += valor;
      console.log(`Saldo atualizado: R$ ${this._saldo.toFixed()}`);
    }
  }

  public consultarSaldo(): number {
    return this._saldo;
  }

  public transferir(contaDestino: Conta, valor: number): void {
    this.sacar(valor);
    contaDestino.depositar(valor);
  }

  public mostrarInfo(): void {
    console.log(`
  | ID CONTA: ${this._id}
  | NÚMERO: ${this._numero}
  | DATA ABERTURA: ${this._dataAbertura}
  | TITULAR: ${this._titular.mostrarDados()}
  | SALDO: ${this._saldo.toFixed(2)}
  ---------------------------------------`);
  }

  public get id(): number {
    return this._id;
  }

  public set id(id: number) {
    this._id = id;
  }

  public get numero(): string {
    return this._numero;
  }

  public set numero(numero: string) {
    this._numero = numero;
  }

  public get saldo(): number {
    return this._saldo;
  }

  public set saldo(saldo: number) {
    this._saldo = saldo;
  }

  public get titular(): Cliente {
    return this._titular;
  }

  public set titular(cliente: Cliente) {
    this._titular = cliente;
  }

  public get dataAbertura(): string {
    return this._dataAbertura;
  }

  public set dataAbertura(dataAbertura: string) {
    this._dataAbertura = dataAbertura;
  }
}
